Mathlib/Data/Polynomial/AlgebraMap.lean

```diff
@@ -164,77 +167,87 @@ variable (x : A)
 the unique `R`-algebra homomorphism from `R[X]` to `A` sending `X` to `x`.
 
-This is a stronger variant of the linear map `polynomial.leval`. -/
+This is a stronger variant of the linear map `Polynomial.leval`. -/
 def aeval : R[X] →ₐ[R] A :=
-  { eval₂RingHom' (algebraMap R A) x fun a => Algebra.commutes _ _ with
-    commutes' := fun r => eval₂_C _ _ }
+  { eval₂RingHom' (algebraMap R A) x fun _a => Algebra.commutes _ _ with
+    commutes' := fun _r => eval₂_C _ _ }
 #align polynomial.aeval Polynomial.aeval
 
-variable {R A}
+-- porting note: removed `variable` due to redundant binder annotation update
 
 @[simp]
-theorem adjoin_x : Algebra.adjoin R ({X} : Set R[X]) = ⊤ := by
-  refine' top_unique fun p hp => _
+theorem adjoin_X : Algebra.adjoin R ({X} : Set R[X]) = ⊤ := by
+  refine' top_unique fun p _hp => _
   set S := Algebra.adjoin R ({X} : Set R[X])
   rw [← sum_monomial_eq p]; simp only [← smul_X_eq_monomial, Sum]
-  exact S.sum_mem fun n hn => S.smul_mem (S.pow_mem (Algebra.subset_adjoin rfl) _) _
-#align polynomial.adjoin_X Polynomial.adjoin_x
+  exact S.sum_mem fun n _hn => S.smul_mem (S.pow_mem (Algebra.subset_adjoin rfl) _) _
+set_option linter.uppercaseLean3 false in
+#align polynomial.adjoin_X Polynomial.adjoin_X
 
-@[ext]
+@[ext 1002]
 theorem algHom_ext {f g : R[X] →ₐ[R] A} (h : f X = g X) : f = g :=
-  AlgHom.ext_of_adjoin_eq_top adjoin_x fun p hp => (Set.mem_singleton_iff.1 hp).symm ▸ h
+  AlgHom.ext_of_adjoin_eq_top adjoin_X fun _p hp => (Set.mem_singleton_iff.1 hp).symm ▸ h
 #align polynomial.alg_hom_ext Polynomial.algHom_ext
 
 theorem aeval_def (p : R[X]) : aeval x p = eval₂ (algebraMap R A) x p :=
   rfl
 #align polynomial.aeval_def Polynomial.aeval_def
 
-@[simp]
+-- porting note: removed `@[simp]` because `simp` can prove this
 theorem aeval_zero : aeval x (0 : R[X]) = 0 :=
   AlgHom.map_zero (aeval x)
 #align polynomial.aeval_zero Polynomial.aeval_zero
 
-@[simp]
-theorem aeval_x : aeval x (X : R[X]) = x :=
+theorem aeval_X : aeval x (X : R[X]) = x :=
   eval₂_X _ x
-#align polynomial.aeval_X Polynomial.aeval_x
+set_option linter.uppercaseLean3 false in
+#align polynomial.aeval_X Polynomial.aeval_X
 
 @[simp]
-theorem aeval_c (r : R) : aeval x (C r) = algebraMap R A r :=
+theorem aeval_C (r : R) : aeval x (C r) = algebraMap R A r :=
   eval₂_C _ x
-#align polynomial.aeval_C Polynomial.aeval_c
+set_option linter.uppercaseLean3 false in
+#align polynomial.aeval_C Polynomial.aeval_C
 
 @[simp]
 theorem aeval_monomial {n : ℕ} {r : R} : aeval x (monomial n r) = algebraMap _ _ r * x ^ n :=
   eval₂_monomial _ _
 #align polynomial.aeval_monomial Polynomial.aeval_monomial
 
-@[simp]
-theorem aeval_x_pow {n : ℕ} : aeval x ((X : R[X]) ^ n) = x ^ n :=
-  eval₂_X_pow _ _
-#align polynomial.aeval_X_pow Polynomial.aeval_x_pow
+@[simp] -- porting note: simplifed the LHS because of `simpNF`
+theorem aeval_X_pow {n : ℕ} : aeval x (X : R[X]) ^ n = x ^ n := by
```

Comment:
I think it would make morme sense to remove `simp` attribute and leave the original lemma as is.

Reply:
done